guides/keras_cv/cut_mix_mix_up_and_rand_augment.py

```diff
@@ -0,0 +1,395 @@
+    value_range=(0, 255), magnitude=0.75, magnitude_stddev=0.3
+)
+
+"""
+First, let's filter out `RandomRotation` layers
+"""
+
+layers = [
+    layer for layer in layers if not isinstance(layer, keras_cv.layers.RandomRotation)
+]
+
+"""
+Next, let's add `GridMask` to our layers:
+"""
+
+layers = layers + [keras_cv.layers.GridMask()]
+
+"""
+Finally, we can put together our pipeline
+"""
+
+pipeline = keras_cv.layers.RandomAugmentationPipeline(
+    layers=layers, augmentations_per_image=3
+)
+
+"""
+Let's check out the results!
+"""
+
+
+def apply_pipeline(inputs):
+    inputs["images"] = pipeline(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(apply_pipeline, num_parallel_calls=AUTOTUNE)
+visualize_dataset(train_dataset, title="After custom pipeline")
+
+"""
+Awesome! As you can see, no images were randomly rotated. You can customize the
+pipeline however you like:
+"""
+
+pipeline = keras_cv.layers.RandomAugmentationPipeline(
+    layers=[keras_cv.layers.GridMask(), keras_cv.layers.Grayscale(output_channels=3)],
+    augmentations_per_image=1,
+)
+
+"""
+This pipeline will either apply `GrayScale` or GridMask:
+"""
+
+
+def apply_pipeline(inputs):
+    inputs["images"] = pipeline(inputs["images"])
+    return inputs
+
+
+train_dataset = load_dataset().map(apply_pipeline, num_parallel_calls=AUTOTUNE)
+visualize_dataset(train_dataset, title="After custom pipeline")
+
+"""
+Looks great! You can use `RandomAugmentationPipeline` however you want.
+"""
+
+"""
+## Training a CNN
+
+As a final exercise, let's take some of these layers for a spin. In this section, we
+will use `CutMix`, `MixUp`, and `RandAugment` to train a state of the art `ResNet50`
+image classifier on the Oxford flowers dataset.
+
+"""
+
+
+def preprocess_for_model(inputs):
+    images, labels = inputs["images"], inputs["labels"]
+    images = tf.cast(images, tf.float32)
+    return images, labels
+
+
+train_dataset = (
+    load_dataset()
+    .map(apply_rand_augment, num_parallel_calls=AUTOTUNE)
+    .map(cut_mix_and_mix_up, num_parallel_calls=AUTOTUNE)
+)
+
+visualize_dataset(train_dataset, "CutMix, MixUp and RandAugment")
+
+train_dataset = train_dataset.map(preprocess_for_model, num_parallel_calls=AUTOTUNE)
+
+test_dataset = load_dataset(split="test")
+test_dataset = test_dataset.map(preprocess_for_model, num_parallel_calls=AUTOTUNE)
+
+train_dataset = train_dataset.prefetch(AUTOTUNE)
+test_dataset = test_dataset.prefetch(AUTOTUNE)
+
+train_dataset = train_dataset.repeat()
+test_dataset = test_dataset.repeat()
+
+"""
+Next we should create a the model itself. Notice that we use `label_smoothing=0.1` in
+the loss function. When using `MixUp`, label smoothing is _highly_ recommended.
+"""
+
+input_shape = IMAGE_SIZE + (3,)
+
+
+def get_model():
+    inputs = keras.layers.Input(input_shape)
+    x = applications.ResNet50V2(
+        input_shape=input_shape, classes=num_classes, weights=None
+    )(inputs)
+    model = keras.Model(inputs, x)
```

Comment:
It's nice to make it more explicit as follows

```python
keras.Model(inputs=inputs, outputs=outputs)
```

Also, the variable `x` can be renamed with a more descriptive word here, as an output.